### PR TITLE
feat: add organization limits

### PR DESCRIPTION
**Pull request by JivusAyrus**
- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

COMPLETES ENG-4495
COMPLETES ENG-4536
FIXES ENG-4510

### CODE REVIEW

File: controlplane/src/core/repositories/analytics/MonthlyRequestViewRepository.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const res = await this.client.queryPromise(query);
+
+    if (Array.isArray(res)) {
+      return res[0].totalRequests;
+    }
+
+    return Number(res);
+  }
+
+  public async getMonthlyRequestSeries(organizationId: string): Promise<PlainMessage<RequestSeriesItem>[]> {
```
[the remark this answers is not in the document]
Comment (JivusAyrus) [reply controlplane/src/core/repositories/analytics/MonthlyRequestViewRepository.ts:40]: yep, its not required, just kept it for future use, if its necessary

File: controlplane/src/core/repositories/OrganizationRepository.ts [hunk 2]
```diff
@@ -136,6 +145,7 @@ export class OrganizationRepository {
           userID: input.userId,
           organizationID: org.id,
         }),
+        limits: await this.getOrganizationLimits({ organizationID: org.id }),
```
Comment (StarpTech) [inline controlplane/src/core/repositories/OrganizationRepository.ts:148]: Can we fetch the limits in the organization query?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: controlplane/src/core/repositories/OrganizationRepository.ts (hunk 2 above)
Comment (JivusAyrus) [reply controlplane/src/core/repositories/OrganizationRepository.ts:148]: you mean a join right?, yep we can

File: controlplane/src/core/repositories/analytics/MonthlyRequestViewRepository.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JivusAyrus) [reply controlplane/src/core/repositories/analytics/MonthlyRequestViewRepository.ts:40]: Removed

File: controlplane/src/core/repositories/OrganizationRepository.ts (hunk 2 above)
Comment (StarpTech) [reply controlplane/src/core/repositories/OrganizationRepository.ts:148]: Yes

File: controlplane/src/core/bufservices/PlatformService.ts [hunk 3]
```diff
@@ -1102,13 +1123,28 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
           };
         }
 
-        const subgraphRepo = new SubgraphRepository(opts.db, authContext.organizationId);
+        const dateRange = {
+          start: req.startDate,
+          end: req.endDate,
+        };
+
+        const orgLimits = await orgRepo.getOrganizationLimits({ organizationID: authContext.organizationId });
```
[the remark this answers is not in the document]
Comment (JivusAyrus) [reply controlplane/src/core/bufservices/PlatformService.ts:1119]: Done

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**